lib/modules/versioning/debian/index.spec.ts

```diff
@@ -0,0 +1,376 @@
+  `(
+    'getSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.getSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  it.each`
+    versions                                            | range         | expected
+    ${['8', '9', '10', '11']}                           | ${'2020.04'}  | ${null}
+    ${['8', '9', '10', '11']}                           | ${'foobar'}   | ${null}
+    ${['8', '9', '10', '11']}                           | ${'11'}       | ${'11'}
+    ${['8', '9', '10', '11']}                           | ${'10'}       | ${'10'}
+    ${['8', '9', '10', '11']}                           | ${'4'}        | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'2020.04'}  | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'foobar'}   | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'bullseye'} | ${'bullseye'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'buster'}   | ${'buster'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'warty'}    | ${null}
+    ${['jessie', 'stretch', 'buster', 'stable']}        | ${'stable'}   | ${'stable'}
+    ${['jessie', 'stretch', 'oldstable', 'bullseye']}   | ${'buster'}   | ${'oldstable'}
+    ${['jessie', 'oldoldstable', 'buster', 'bullseye']} | ${'warty'}    | ${null}
+  `(
+    'minSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.minSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  it.each`
+    currentValue      | rangeStrategy | currentVersion | newVersion    | expected
+    ${undefined}      | ${undefined}  | ${undefined}   | ${'foobar'}   | ${'foobar'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'11'}       | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'bullseye'} | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'stable'}   | ${'bullseye'}
+    ${'9'}            | ${undefined}  | ${undefined}   | ${'11'}       | ${'11'}
+    ${'oldoldstable'} | ${undefined}  | ${undefined}   | ${'11'}       | ${'stable'}
+    ${'oldstable'}    | ${undefined}  | ${undefined}   | ${'11'}       | ${'stable'}
+    ${'9'}            | ${undefined}  | ${undefined}   | ${'stable'}   | ${'11'}
+    ${'oldstable'}    | ${undefined}  | ${undefined}   | ${'3'}        | ${'3'}
+    ${'oldstable'}    | ${'pin'}      | ${undefined}   | ${'11'}       | ${'11'}
+    ${'oldstable'}    | ${'pin'}      | ${undefined}   | ${'stable'}   | ${'11'}
+    ${'oldstable'}    | ${'pin'}      | ${undefined}   | ${'bullseye'} | ${'11'}
+    ${'Buster'}       | ${'pin'}      | ${undefined}   | ${'11'}       | ${'11'}
+    ${'Buster'}       | ${'pin'}      | ${undefined}   | ${'stable'}   | ${'11'}
+    ${'Buster'}       | ${'pin'}      | ${undefined}   | ${'bullseye'} | ${'11'}
```

Comment:
i don't thiunk we need those tests, as this function should ever only called with `valid` versions.

or is it a typo `Buster`=> `buster` ?

Reply:
typo, and indeed an invalid input that should be filtered out.
added this as a test case in `isValid` and also ran against a real repo just to be sure.

heres the relevant log of this run, 

```
DEBUG: Dependency debian has unsupported value Buster (repository=ladzaretti/docker-debian)
```
and 
```
           }
             "packageFile": "3/Dockerfile",
             "deps": [
               {
                 "depName": "debian",
                 "currentValue": "Buster",
                 "replaceString": "debian:Buster",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "versioning": "debian",
                 "depType": "final",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "skipReason": "invalid-value"
               }
             ]
           }